pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -227,6 +247,29 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 		}
 	}
 
+	desiredMetric := resources.MakeMetric(ctx, pa, c.dynConfig.Current())
+	metric, err := c.metrics.Get(ctx, desiredMetric.Namespace, desiredMetric.Name)
```

Comment:
It seems like theres a narrow race since we are releasing the c.metrics lock between here and create. Should we have a GetOrCreate method in Metrics which can hold the lock for both of these operations?

Reply:
This is modeled after how the Deciders work which is generally modeled after how entities that are persisted in K8s work in general. The idea is that it is fairly easily possible to eventually pull this out into a full blown CRD (which might happen to this in the future).

In that sense, I'd like to keep this parallel to how the decider works.

Reply:
Ah, and theres no race because we guard against that during create, now it makes sense :)

Since this is unintuitive could we drop a comment about that here? something like
```
// although create would be sufficient we are modeling CRD api usage here
```